coralproject/talk
# [CORL-2523]: propagate jwt verification errors properly

## What does this PR do?

These changes improve the error thrown if no suitable jwt verifier is found by clarifying that it could be due to a verifier not being enabled or to validation errors. If there are validation errors, those are included along with which verifier they are for. Interested in any input into the error messages--their wording and how they include the errors--to make sure they clarify the potential issues and better help people resolve them.

## What changes to the GraphQL/Database Schema does this PR introduce?

none

## Does this PR introduce any new environment variables or feature flags? 

no

## If any indexes were added, were they added to `INDEXES.md`?

n/a

## How do I test this PR?

You can test this by making some errors happen and trying different combinations of SSO/OIDC/local being enabled. You can make some errors happen by just updating the token in the code (removing required fields, etc.). See that if multiple auth integrations are enabled, as long as one has a valid token, there will be no error. However, if none have a valid token, then all token validation errors are included in the error context.
 
## How do we deploy this PR?
